kernel/src/syscall/mmap.rs

```diff
@@ -57,12 +57,22 @@ fn do_sys_mmap(
     if len == 0 {
         return_errno_with_message!(Errno::EINVAL, "mmap len cannot be zero");
     }
+    if len > isize::MAX as usize {
+        return_errno_with_message!(Errno::ENOMEM, "mmap len too large");
+    }
 
     let len = len.align_up(PAGE_SIZE);
 
     if offset % PAGE_SIZE != 0 {
         return_errno_with_message!(Errno::EINVAL, "mmap only support page-aligned offset");
     }
+    offset.checked_add(len).ok_or(Error::with_message(
+        Errno::EOVERFLOW,
+        "integer overflow when (offset + len)",
+    ))?;
+    if addr > isize::MAX as usize - len {
+        return_errno_with_message!(Errno::ENOMEM, "mmap (addr + len) too large");
+    }
```

Comment:
This should also be EINVAL. In [man page](https://man7.org/linux/man-pages/man2/mmap.2.html):
>  EINVAL We don't like addr, length, or offset (e.g., they are to large, or not aligned on a page boundary)

I think you use ENOMEM here because this:
> ENOMEM We don't like addr, because it exceeds the virtual address space of the CPU.

I'm not very sure, but I think this case should be `addr > MAX_USER_SPACE_ADDR`, and `addr + len > isize::MAX` is more like an overflow error. Have you tried these two cases in Linux? which errors do they return?

Reply:
I use `ENOMEM` here because in https://elixir.bootlin.com/linux/v6.10.5/source/mm/mmap.c#L1248.
Linux uses the `ENOMEM` to report the overflow error, and I don't know why...
```C
/* Careful about overflows.. */
	len = PAGE_ALIGN(len);
	if (!len)
		return -ENOMEM;
```

Reply:
As for the `addr > MAX_USER_SPACE_ADDR` condition, linux also uses the `ENOMEM` in https://elixir.bootlin.com/linux/v6.10.5/source/mm/mmap.c#L1896

```C
if (addr > TASK_SIZE - len)
	return -ENOMEM;
```

Reply:
OK, let's keep it consistent with Linux.